### PR TITLE
Update 'gateway-plugin' to work with JetBrains Gateway v222.3345.1+

### PR DESCRIPTION
**Pull request by felladrin**
## Description

Update 'gateway-plugin' to work with JetBrains Gateway v222.3345.1+.

image

The main difference is that v222.3345.1 had a change on the constructor parameters of `ClientOverSshTunnelConnector`.

Some other small updates/renames were made to the code to resolve issues like deprecation and variable shadowing, pointed out by IntelliJ IDEA.

## Related Issue(s)

None.

## How to test

- Download JetBrains Gateway v222.3345.1 for Mac from [here](https://gitpod.slack.com/archives/C02BRJLGPGF/p1657041136854829) (internal link), and install it.
- [Download this Gateway Plugin build, from Gitpod's Plugin Dev Channel](https://plugins.jetbrains.com/plugin/18438-gitpod-gateway/versions/Dev/193675), and [install it on the Gateway](https://www.jetbrains.com/help/idea/managing-plugins.html#install_plugin_from_disk).
- Confirm if you can connect to any workspace via using JetBrains Gateway v222.3345.1+.
  image

## Release Notes

```release-note
Add support for JetBrains Gateway v222.3345.1 and later.
```

## Documentation

None.

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: components/ide/jetbrains/gateway-plugin/gradle.properties [hunk 1]
```diff
@@ -6,15 +6,15 @@ pluginName=gitpod-gateway
...
+pluginSinceBuild=222.3325
 pluginUntilBuild=222.*
 # Plugin Verifier integration -> https://github.com/JetBrains/gradle-intellij-plugin#plugin-verifier-dsl
 # See https://jb.gg/intellij-platform-builds-list for available build versions.
 pluginVerifierIdeVersions=2022.1, 2022.2
 # IntelliJ Platform Properties -> https://plugins.jetbrains.com/docs/intellij/tools-gradle-intellij-plugin.html#intellij-extension-type
 platformType=GW
 # Version from "com.jetbrains.gateway" which can be found at https://www.jetbrains.com/intellij-repository/snapshots
-platformVersion=222.3244.1-CUSTOM-SNAPSHOT
+platformVersion=222-NIGHTLY-CUSTOM-SNAPSHOT
```
Comment (felladrin) [inline components/ide/jetbrains/gateway-plugin/gradle.properties:17]: TODO: Update it to a fixed version when JetBrains team release a new Gateway Snapshot at https://www.jetbrains.com/intellij-repository/snapshots

The last version published is '222.3244.1-CUSTOM-SNAPSHOT', which doesn't have the `SshHostTunnelConnector` class.

After that, this PR will be ready for review.

Comment (felladrin) [reply components/ide/jetbrains/gateway-plugin/gradle.properties]: Updated.

**Comment by akosyakov** [comment]
@felladrin what is blocking landing this PR?

**Comment by felladrin** [comment]
> @felladrin what is blocking landing this PR?

Just the approval of a reviewer.

**Review by loujaybee — APPROVED** [review]

**Pull request merged**